### DISCUSSION TITLE
How to reset a date picker after form submission

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#8100] [Q&A]
Upon successful form submission I don't redirect the user, I just want to empty the form. 

Using `e.target.reset();` works for all fields I have except the date picker.

I have tried `setValue(tripDate, "");` and  `reset({ ...currentFormValues, tripDate: "" });` but both of these seem to crash the application. 

```
            <Controller
              control={control}
              name="tripDate"
              render={({ field }) => (
                <DatePicker
                  name="tripDate"
                  placeholderText="Select date"
                  onChange={(e) => field.onChange(e)}
                  selected={field.value}
                  dateFormat="dd/MM/Y"
                  minDate={new Date()}
    
                />
              )}
            />
```

**Comment by Moshyfawn** [comment]
You will have to provide a minimal reproduction codesanbox with the issue you're facing for us to be able to help you

**Comment by ritikbanger** [comment]
Update this:  `render={({ field: { value, onChange } })` or make sure that you have the field `tripDate` in your default values because reset works to reset the fields to its default value. I assume, you have not provided default value for `tripDate` is the issue.

**Comment by ghost** [comment]
Actually, I managed to get it working by just using reset() so I can close this out now :)

**Reply by sidneywidmer** [reply]
How?
